tortoise/README.md

```diff
@@ -0,0 +1,167 @@
+
+###### Base ballot and exceptions
+
+Discussed in [Voting section](#voting).
+
+###### Weight
+
+Weight of the ballot is computed from the weight of the activation transaction divided by the total number of eligible ballots. Total number of eligible ballots is computed based on the fraction of actiovation transaction compared to the weight from the active set that was recorded in reference ballot.
+
+###### Beacon
+
+Ballot with an inconsistent beacon will not be marked good and will not be counted immediatly. Discussed in [full](#votes-from-ballots-with-wrong-beacon) and [verifying](#good-ballots) sections.
+
+#### Expected weight and thresholds
+
+Expected weight is a total weight from the activation transactions. Unlike weight for the ballot we are using a sum of all activation transactions that target a specific epoch. Expected weight for the layer is fraction of the expected weight for the epoch.
+
+## Voting
+
+When voting protocol chooses base ballot and encodes the difference between local opinion and base ballot votes. Since we don't keep votes from genesis in memory the implementation has a shortcoming that it encodes difference only within some configurable window (we can refer to it as voting window). 
+
+In order to minimize amount of data that needs to be sent and stored, Tortoise selects ballot that has the least difference with a local opinion. 
+
+Types of votes:
+- support
+
+    Support all blocks according to local opinion that are not supported by base ballot. Usually support is added for layers starting from base ballot layer up to last layer. In the event of healing tortoise may add support for blocks before base ballot layer.
+
+- against
+  
+    Against is added only if the base ballot supports the block and local opinion is against that block. 
+
+- abstain (undecided) 
+    
+    Undecided is a valid vote only until hare is terminated for the layer (it may take more than one layer for hare to terminate, this is expressed as zdist).
+
+If ballot doesn't specify explicit vote for a block then it votes agains the block. This prevents malicious smeshers from retroactively reorganizing mesh by intentionally creating late ballots.
+
+There are 4 distinct reasons to cast a vote:
+- hare output
+    
+    tortoise uses output from the hare consensus for voting on blocks within hdist.
+
+- validity/decisions
+
+    for layer before hdist the vote is casted according to the decision. 
+
+- local threshold
+
+    in case when decision wasn't reached but margin from counted weight crosses local threshold tortoise will cast a vote according to the local threshold.
+
+- weak coin
+
+    otherwise tortoise votes according to the coin recorded in the last layer. if sufficient number of honest votes were casted this way then tortoise will either reach decision or atleast vote according to the local threshold in future layers. if honest nodes disagreed on the coin value then the voting will continue until they agree.
```

Comment:
```suggestion
    otherwise tortoise votes according to the coin recorded in the last layer. if sufficient number of honest votes were cast this way then tortoise will either reach decision or at least vote according to the local threshold in future layers. if honest nodes disagreed on the coin value then the voting will continue until they agree.
```